src/services/fileServices/MdPageServices/PageService.ts

```diff
@@ -214,62 +214,56 @@ export class PageService {
   // The page can be nested 1 or 2 levels deep:
   // eg: one/level or two/levels/deep
   private parseResourceRoomPage = (
-    pageName: string,
+    pathInfo: PathInfo,
     sessionData: UserSessionData
   ): ResultAsync<
     ResourceCategoryPageName,
     NotFoundError | MissingResourceRoomError
   > =>
     this.extractResourceRoomName(sessionData).andThen((name) =>
-      this.extractResourcePageName(name, pageName, sessionData)
+      this.extractResourcePageName(name, pathInfo, sessionData)
     )
 
   private extractResourcePageName = (
     resourceRoomName: ResourceRoomName,
-    pageName: string,
+    { name, path }: PathInfo,
     sessionData: UserSessionData
   ): ResultAsync<ResourceCategoryPageName, NotFoundError> =>
-    this.extractPathInfo(pageName)
-      .asyncAndThen(({ name, path }) =>
-        path.asyncAndThen<ResourceCategoryPageName, NotFoundError>(
-          (rawPath) => {
-            if (rawPath[0] !== resourceRoomName.name) {
-              return errAsync(new NotFoundError())
-            }
+    path
+      .asyncAndThen<ResourceCategoryPageName, NotFoundError>((rawPath) => {
+        if (rawPath[0] !== resourceRoomName.name) {
+          return errAsync(new NotFoundError())
+        }
 
-            if (rawPath.length !== 3 && rawPath.at(-1) !== "_posts") {
-              return errAsync(new NotFoundError())
-            }
+        if (rawPath.length !== 3 && rawPath.at(-1) !== "_posts") {
```

Comment:
May I get your help to add a comment of how a `rawPath` might look like? abit confused what the each element of `rawPath` represents

Reply:
can but i'll add it onto the `PathInfo` type - this is to avoid repetition + set at the containing type. wdyt?

Reply:
can but i'll add it onto the `PathInfo` type - this is to avoid repetition + set at the containing type. wdyt?